Dockerfile_mbtci_java17

```diff
@@ -1,93 +1,189 @@
+      | cut -d: -f1 \
+      | sort -u \
+      | xargs -r apt-mark manual \
+    && apt-get purge -y --auto-remove -o APT::AutoRemove::RecommendsImportant=false \
+    && ln -s /usr/local/bin/node /usr/local/bin/nodejs \
+    # smoke tests
+    && echo "node ${NODE_VERSION} install smoke tests!" \
+    && node --version \
+    && npm --version
 
-# installing Golang
-	curl -O https://storage.googleapis.com/golang/go${GO_VERSION}.linux-amd64.tar.gz && tar -xvf go${GO_VERSION}.linux-amd64.tar.gz && \
-    mv go /usr/local && \
-    mkdir -p "$GOPATH/src" "$GOPATH/bin" && chmod -R 777 "$GOPATH" && \
-    mkdir -p ${GOPATH}/src ${GOPATH}/bin && \
+# install ui5-cli temporay solution
+RUN	set -ex \
+    && npm install --prefix /usr/local/ -g @ui5/cli \
+    && npm cache clean -g --force \
+    # smoke tests
+    && echo "ui5 install smoke tests!" \
+    && ui5 --version
 
-# update maven home
-	M2_BASE="$(dirname ${M2_HOME})" && \
-    mkdir -p "${M2_BASE}" && \
-    curl --fail --silent --output - "https://apache.osuosl.org/maven/maven-3/${MAVEN_VERSION}/binaries/apache-maven-${MAVEN_VERSION}-bin.tar.gz" \
-    | tar -xzvf - -C "${M2_BASE}" && \
-    ln -s "${M2_HOME}/bin/mvn" /usr/local/bin/mvn && \
-    chmod --recursive a+w "${M2_HOME}"/conf/* && \
+# install Golang
+RUN	set -ex \
+    && apt-get update \
+    && apt-get install -y ca-certificates curl --no-install-recommends \
+    && rm -rf /var/lib/apt/lists/* \
+    && curl -O https://storage.googleapis.com/golang/go${GO_VERSION}.linux-amd64.tar.gz \
+    && tar -xvf go${GO_VERSION}.linux-amd64.tar.gz \
+    && rm -rf /usr/local/go \
+    && mv go /usr/local \
+    && mkdir -p "$GOPATH/src" "$GOPATH/bin" \
+    && chmod -R 777 "$GOPATH" \
+    && mkdir -p ${GOPATH}/src ${GOPATH}/bin \
+    # smoke tests
+    && echo "go ${GO_VERSION} install smoke tests!" \
+    && go version
 
-# Download MBT
-	curl -L "https://github.com/SAP/cloud-mta-build-tool/releases/download/v${MBT_VERSION}/cloud-mta-build-tool_${MBT_VERSION}_Linux_amd64.tar.gz" | tar -zx -C /usr/local/bin && \
-    chown root:root /usr/local/bin/mbt && \ 
+# install sapmachine
+RUN ARCH= && dpkgArch="$(dpkg --print-architecture)" \
+    && case "${dpkgArch##*-}" in \
+      amd64) ARCH='amd64';; \
+      *) echo "unsupported architecture"; exit 1 ;; \
+    esac \
+  && set -ex \
+  && apt-get update \
+  && apt-get install -y --no-install-recommends wget ca-certificates gnupg \
+  && export GNUPGHOME="$(mktemp -d)" \
+  && wget -q -O - https://dist.sapmachine.io/debian/sapmachine.old.key | gpg --batch --import \
+  && gpg --batch --export --armor 'DA4C 00C1 BDB1 3763 8608 4E20 C7EB 4578 740A EEA2' > /etc/apt/trusted.gpg.d/sapmachine.old.gpg.asc \
+  && wget -q -O - https://dist.sapmachine.io/debian/sapmachine.key | gpg --batch --import \
+  && gpg --batch --export --armor 'CACB 9FE0 9150 307D 1D22 D829 6275 4C3B 3ABC FE23' > /etc/apt/trusted.gpg.d/sapmachine.gpg.asc \
+  && gpgconf --kill all \
+  && rm -rf "$GNUPGHOME" \
```

Comment:
Please take the code in mbt-docker internal repo to install SAPMachine, the original documentation do it not in an optimized way.

Reply:
Do you mean switch to the implementation in  "https://github.tools.sap/E-Mobility/mbt-docker/blob/main/Dockerfile" which install sapmachine, right ?